### DISCUSSION TITLE
Feedback/ideas for FlowLauncher as a new user

### DISCUSSION THREAD

**Discussion by bytenekos** [Flow-Launcher/Flow.Launcher#3018] [Ideas]
Hi there! Recently started using flow launcher and thought that I'd voice some of my thoughts/ideas about it to (hopefully) make it more user friendly

Plugin installs:
- Would be nice to have a queue or something because having the app keep restarting after every plugin install is annoying
(You gotta keep scrolling back to where you were after installing a plugin which is annoying.)

- Seems like python/nodejs installs don't get automatically picked up? And after clicking on "no" when it prompts me to install python/nodejs so that I can choose my python/nodejs install location and clicking on cancel it seems like it just still auto installs python/nodejs, which feels weird

- Plugin installer/manager settings are unintuitive. Didn't realise that to disable auto restart flow launcher when installing a new plugin you have to disable it from the plugin settings page (which I guess fixed my first point but still). I would've expected for it to be in the general settings or something like that.
![image](https://github.com/user-attachments/assets/3a9c483c-b4e6-4077-9856-3bca6196df1b)

Shell plugin:
- Shell plugin should have the ability to use custom terminals instead of just the listed ones (I use wezterm for example and I'd like it to open that). I'm not sure if I should be mentioning this here but it seems like it's a default plugin so I thought I'd just mention it here.

Misc:
- Seems like VBS scripts don't... launch when pressing enter on them?

https://github.com/user-attachments/assets/ce2cb849-d6ce-459b-b357-8004657d8002

**Comment by jjw24** [comment]
Hi there, thank you for the valuable feedback. 

>Would be nice to have a queue or something because having the app keep restarting after every plugin install is annoying
(You gotta keep scrolling back to where you were after installing a plugin which is annoying.)

Good point. I think we should change the default behaviour to disable auto restart flow. I will create an issue for this.

>Plugin installer/manager settings are unintuitive. Didn't realise that to disable auto restart flow launcher when installing a new plugin you have to disable it from the plugin settings page (which I guess fixed my first point but still). I would've expected for it to be in the general settings or something like that.

Makes sense, but not sure where to place it yet, whether in the plugin store tab and together with a restart flow button or somewhere in the general settings tab.

**Reply by jjw24** [reply]
Issue created to disable auto restart by default https://github.com/Flow-Launcher/Flow.Launcher/issues/3028

**Comment by jjw24** [comment]
>Seems like python/nodejs installs don't get automatically picked up? 

If you installed them yourself somewhere on the computer they won't be automatically picked up.

>And after clicking on "no" when it prompts me to install python/nodejs so that I can choose my python/nodejs install location and clicking on cancel it seems like it just still auto installs python/nodejs, which feels weird

That's the intended behaviour because most plugins will require python or nodejs,  if you have chosen to select your own install location and then cancel out of the dialogue, flow will go ahead and install them for you otherwise those plugins will error out and we don't want that to happen.

I am open to better UX ideas though around this.

**Comment by jjw24** [comment]
>Shell plugin should have the ability to use custom terminals instead of just the listed ones (I use wezterm for example and I'd like it to open that). I'm not sure if I should be mentioning this here but it seems like it's a default plugin so I thought I'd just mention it here.

I believe there is already an issue for this :)

**Comment by jjw24** [comment]
>Seems like VBS scripts don't... launch when pressing enter on them?

If this is the case please create a bug issue :)
